### ISSUE TITLE
Flawed metadata support

### ISSUE THREAD

**Issue by stof** [KnpLabs/Gaufrette#108]
In the Amazon adapter, metadata are used from memory when renaming a file, but the metadata are never populated from the persistent ones.
The same issue exists for ``getMetadata``: it will return them only in the process defining them as it won't read them from amazon

**Comment by wysow** [comment]
Closing this as it seems outdated now...

**Comment by stof** [comment]
@wysow this is not outdated. The metadata handling has not been changed since 2012 and is still flawed.

**Comment by NiR-** [comment]
Wow.. I wasn't aware of this issue until @zyphlar created #451. I will try to deep dive into @TomiS's branch and see what needs extrawork.

**Comment by TomiS** [comment]
@NiR- Yeah, unfortunately my refactoring attempts are pretty much dead code after all these years :)

**Comment by jspizziri** [comment]
@NiR- , did https://github.com/KnpLabs/Gaufrette/pull/451 resolve & close this issue?

**Comment by grempa** [comment]
Are there any new updates to this issue?
